htdocs/fourn/ajax/getSupplierPrices.php

```diff
@@ -86,12 +86,46 @@
 
 	// Add price for costprice
 	$price=$producttmp->cost_price;
+	if (empty($price))
```

Comment:
This may break the fallback and rule 
"if cost price not defined on product, we use AWP price. Because if there is a cost price on subproduct, this will define a cost price on product where there is no cost price, so the fallback on AWP price will not be executed.
Because this offer another rule on how the default price is defined, we must introduce a new option to be able to choose the rule we want. Can you add a condition on a hidden option here:
 && ! empty($conf->global->PRODUCT_USE_SUB_COST_PRICES_IF_COST_PRICE_EMPTY)

Reply:
@eldy i'm not sure... this script returns one entry for price, one for costprice and one for AWP, so defining costprice will not affect AWP

Reply:
Yes, but later in javascript, we use this array return to apply the rule defined into dolibarr setup to say
"if cost price empty, fallback to pmp, if awp empty, fallback onto best supplier price"

Reply:
ok it's done.

Reply:
Last question : If PMP (AWP) or Cost Price 
of one subproduct changes, 
does it reflect immediately on cost price of father, 
or there is an "update" method ?
Thanks

Reply:
it's recalculated, so no need to update ;-)

Reply:
but if the line is already in the document, the cost price is not updated automatically.